### PR TITLE
Drop in Unidata matplotlib workshop notebook

### PR DESCRIPTION
**Pull request by dcamron**
This is a drop-in of [this notebook](https://github.com/Unidata/python-training/blob/master/pages/workshop/Matplotlib/Matplotlib%20Basics.ipynb) from Unidata's workshop materials with exercises modified and branding removed and some minor image updates. Otherwise mostly content-unmodified.

Notes:
 * Looks pretty good, organized similarly to others
 * Relatively small/simple scope, little crossover to other applications. Might fit in to a larger unit or serves as decent intro
 * As with some of the others I've dropped in, ties in pretty heavily to Meteorology concepts/jargon

### CODE REVIEW

**Comment by brian-rose** [comment]
I'm not on the reviewers list, but I took a quick look and there's great content here!

Two quick issues:
- The "Notes" and other admonitions are rendering weirdly in the book
- Would it be possible to break up the content into a few more major section headings `##`? Currently essentially the whole content falls under one heading "Plotting with Matplotlib"

File: core/matplotlib/matplotlib.ipynb [hunk 1]
```diff
@@ -0,0 +1,1192 @@
+{
```
Comment (clyne) [inline core/matplotlib/matplotlib.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    # Ask, out of a 1x1 grid, the first axes.
</pre><p>I find this comment a little confusing. Could you expand?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (clyne) [inline core/matplotlib/matplotlib.ipynb:1]: <p>"facilitate automatic creation" of what? Please clarify what is automatically created.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (clyne) [inline core/matplotlib/matplotlib.ipynb:1]: <p>I'm not seeing any obvious difference between these two plots. Can you clarify how they are different?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (clyne) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Perhaps add a link to the GeoCAT examples <a href="https://geocat-examples.readthedocs.io/en/latest/" target="_blank">gallery</a>, which provides hundreds of geoscience-specific plotting examples</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='63'/>

**Comment by clyne** [comment]
Looks great. I just had a few nits.

File: core/matplotlib/matplotlib.ipynb (hunk 1 above)
Comment (ktyle) [reply core/matplotlib/matplotlib.ipynb:1]: <p>"facilitate automatic creation and placement of legend labels" should do nicely.</p>

Comment (ktyle) [reply core/matplotlib/matplotlib.ipynb:1]: <p>Looks like the y-axis also needs to be shared.</p>

Comment (ktyle) [reply core/matplotlib/matplotlib.ipynb:1]: <p>Good idea!</p>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>In (4), replace "imshow" with "image"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Reword first sentence as "Now let's make our first plot with Matplotlib".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>This markdown cell is sufficient to allow for the elimination of the (commented-out) documentation lines in the code cell immediately below.</p><p> </p><p>Rewrite first sentence as:</p><blockquote>Let's create a <code>Figure</code> whose dimensions, if printed out on hardcopy, would be 10 inches wide and 6 inches long (assuming a landscape orientation). We then create an <code>Axes</code>, consisting of a single subplot, on the <code>Figure</code>.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Not crazy about the wording "ask the axis". How about rewording thusly:</p><p> </p><blockquote>Next, add x- and y-axis labels to our <code>Axes</code>  object.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Instead of passing in a <code>fontdict</code>  object, how about simply using "size = 16" as the argument to <code>set_title</code>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>"by" not "be"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Perhaps add an info box, anticipating the inevitable question of how one can deal with a situation as in the above plot, where the legend label (despite being somewhat transparent) obscures part of the traces?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Ddd "aka" before "a scatter plot"</p><p>none --&gt; <code>None</code></p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>methods --&gt; method</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Rewrite so the "Exercise" aspect of this is removed (although in general, we need to think about including "Exercise" code blocks in our notebooks)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>"Let's start" implies that a more complex <code>imshow</code>  example will follow. Either provide such an example, or rewrite "Let's start with" as "Here is".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Add a markdown cell above this that says something to the effect of "Now let's label our contour lines."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='56'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Add a Markdown cell introducing the <code>contourf</code>  method.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>imshow --&gt; <code>imshow</code></p><p> contour --&gt; <code>contour</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='58'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>Make the five examples here consistent with what's at the beginning of the notebook. Perhaps something like:</p><ul><li>Line plots (<code>plot</code>)</li><li> Scatter plots (<code>scatter</code>)</li><li> Image plots (<code>imshow</code>)</li><li> Contour line and contour fill plots (<code>contour</code>,  <code> contourf</code>)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='61'/>

**Comment by clyne** [comment]
@mgrover1 just checking in to see if you think this is still doable for wednesday. Thanks!

**Comment by mgrover1** [comment]
> @mgrover1 just checking in to see if you think this is still doable for wednesday. Thanks!

Yes - sorry I will try to push the new changes today... busy with the CESM workshop this week, but I will get those changes made so you all can take a look and we can get this merged

**Comment by clyne** [comment]
Coolio. Thanks!

**Comment by mgrover1** [comment]
All suggestions have been addressed I believe - would appreciate another look-over from @ktyle and @clyne

**Comment by clyne** [comment]
On it! FYI you can trigger a re-review whenever you are ready by clicking the circular arrows next to the reviewers names.

File: core/matplotlib/matplotlib.ipynb (hunk 1 above)
Comment (clyne) [reply core/matplotlib/matplotlib.ipynb:1]: Agreed. With only 'sharex' there is no discernible difference between the two plots.

Comment (mgrover1) [reply core/matplotlib/matplotlib.ipynb:1]: <p>It now includes both sharex and sharey</p>

Comment (clyne) [reply core/matplotlib/matplotlib.ipynb:1]: Oops. Nevermind. Looks good

**Review by clyne — APPROVED** [review]
This is awesome, Max!

File: core/matplotlib/matplotlib.ipynb [hunk 2]
```diff
@@ -0,0 +1,1214 @@
...
```
Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>independant --&gt; independent</p><p>dependant --&gt; dependent</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (ktyle) [inline core/matplotlib/matplotlib.ipynb:1]: <p>reate --&gt; create</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ProjectPythia/pythia-foundations/pull/37/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='61'/>

**Comment by mgrover1** [comment]
@ktyle I went ahead and fixed the spelling - thanks for the comments!

**Comment by ktyle** [comment]
@mgrover1 great! I think we're just about there! The only other thing I would recommend is that you de-emphasize the "exercise-like" instructions that proceed one of the last notebooks ... maybe just delete the "Try to" in the numbered item that specifies adding a colorbar.

**Comment by mgrover1** [comment]
@ktyle I made that change

**Comment by ktyle** [comment]
LGTM and I think this is ready to merge ...

**Comment by mgrover1** [comment]
@ktyle Awesome! Can you hit the approve button?

**Comment by ktyle** [comment]
@mgrover1 by "Approve" do you mean "Squash and merge"?

**Comment by mgrover1** [comment]
@ktyle Sure - that or the "formal review" at the top of the page? Wasn't sure if we technically needed two approvals to merge

**Review by ktyle — APPROVED** [review]

**Comment by brian-rose** [comment]
Great work all! We're checking things off the milestone list...
